core/explore_api.py

```diff
@@ -258,3 +239,31 @@ def create_spreadsheet(name,refresh_token):
     except Exception as e:
         logger.error(f"Error creating spreadsheet: {e}")
         return e
+
+
+@router.get("/workspaces/{workspace_id}/{explore_id}/status", response={200: str, 400: DetailSchema})
+def get_explore_status(request,workspace_id,explore_id,payload:ExploreStatusSchemaIn):
+    data = payload.dict()
+    try:
+        logger.debug("getting_explore_status")
+        explore = Explore.objects.get(id=explore_id)
+        sync_id = data.get('sync_id')
+        response = requests.get(f"http://valmi-activation:8000/syncs/{sync_id}/status")
+        status = response.text
+        print(status)
+        # if status == 'stopped':
+        #     CODE for re running the sync from backend
+        #     payload = SyncStartStopSchemaIn(full_refresh=True)       
+        #     response = create_new_run(request,workspace_id,sync_id,payload)
+        #     print(response)
+        #     return "sync got failed. Please re-try again"
+        if status == '"running"':
+            return "sync is still running"
+        if status == '"failed"':
+            return "sync got failed. Please re-try again"
+        explore.ready = True
```

Comment:
if explore.ready is True just return status, instead of querying it from activation engine

Reply:
resolved